packages/eslint/src/configs/template.ts

```diff
@@ -0,0 +1,23 @@
+import type { FlatConfig } from '@typescript-eslint/utils/ts-eslint';
+import templateParser from '../template-parser';
+import { rules } from '../rules';
+
+const templateRules: FlatConfig.Rules = {};
+
+Object.entries(rules).forEach(([ruleName, rule]) => {
+  if (rule.meta.docs?.category === 'template') {
+    templateRules[`post/template/${ruleName}`] = 'error';
+  }
+});
+
+export default (plugin: FlatConfig.Plugin): FlatConfig.Config => ({
+  name: 'post/template',
+  files: ['**/*.html'],
+  languageOptions: {
+    parser: templateParser,
+  },
+  plugins: {
+    'post/template': plugin,
+  },
+  rules: templateRules,
+});
```

Comment:
It could be a good idea to split this a little bit up and implement different things in different config objects.
At least this is done by most the flat configs I saw (for example from typescript-eslint).

Almost all of them implemented at least two objects, one for the base config, containing `languageOptions`, `plugins` and other configurations and one to add the `files` filter and the `rules`.

```js
export default (plugin: FlatConfig.Plugin): FlatConfig.Config => ({
  name: 'post/design-system/base',
  languageOptions: {
    parser: templateParser,
  },
  plugins: {
    'post/template': plugin,
  },
},
{
  name: 'post/design-system/recommended',
  files: ['**/*.html'],
  rules: templateRules,
});
```

Let me know, what you think about this idea.

Reply:
Yes, this is the idea. I only implemented the template config for now but we can definitely add more.

What differs from typescript-eslint is that we will have multiple parsers: one for HTML, one for TypeScript and one for Sass. Then, the config object cannot really be split as the template parser only applies to HTML files and HTML files can only be parsed by the template parser.

Reply:
angular-eslint is a more appropriate example for us as they use different parsers as well. Here are their configs: https://github.com/angular-eslint/angular-eslint/tree/main/packages/angular-eslint/src/configs

Reply:
Do you mean using a `ConfigArray`? 
If so we can definitely align with other plugins, do you know what are the benefits of the array over the object? It seems more difficult to override a rule with an array...

Reply:
Since every `ConfigObject` ends up in a `ConfigArray` anyway and acts as an override for the previous `ConfigObject`, this should be save and is (as explained) exaclty what `typescript-eslint`, `angular-eslint` and other packages are doing already. Also splitting up the `files` filter and the `parser` config is not an issue, as following `ConfigObjects` can always override this parameter.